### PR TITLE
Introduce copy_from

### PR DESCRIPTION
**Pull request by xbrianh**

### CODE REVIEW

File: ssds/blobstore/local.py [hunk 1]
```diff
@@ -52,6 +53,14 @@ def put(self, data: bytes):
         with open(self._path, "wb") as fh:
             fh.write(data)
 
+    @catch_blob_not_found
+    def copy_from(self, src_blob: "LocalBlob"):
+        """
+        Intra-cloud copy
+        """
+        assert isinstance(src_blob, type(self))
+        shutil.copyfile(src_blob._path, self._path)
```
Comment (DailyDreaming) [inline ssds/blobstore/local.py:62]: Should this simply return on the edge case where `src_blob._path == self._path`?

File: ssds/blobstore/local.py [hunk 2]
```diff
@@ -52,6 +53,14 @@ def put(self, data: bytes):
         with open(self._path, "wb") as fh:
             fh.write(data)
 
+    @catch_blob_not_found
+    def copy_from(self, src_blob: "LocalBlob"):
```
Comment (DailyDreaming) [inline ssds/blobstore/local.py:57]: Should this be able to copy from `s3` and `gs` to a local path as well?

File: ssds/blobstore/s3.py [hunk 3]
```diff
@@ -73,6 +73,21 @@ def put(self, data: bytes):
         blob = self._s3_bucket.Object(self.key)
         blob.upload_fileobj(io.BytesIO(data))
 
+    def copy_from(self, src_blob: "S3Blob"):
```
Comment (DailyDreaming) [inline ssds/blobstore/s3.py:76]: Same.  Should this only copy between s3 buckets or should this be cloud agnostic?

**Review by DailyDreaming — APPROVED** [review]
LGTM.  See comments, though I expect the answer will be to possibly implement them in a future PR.

File: ssds/blobstore/local.py (hunk 1 above)
Comment (xbrianh) [reply ssds/blobstore/local.py:62]: Will add.

File: ssds/blobstore/local.py (hunk 2 above)
Comment (xbrianh) [reply ssds/blobstore/local.py:57]: This is intended to be "intra-cloud" copy only. Forthcoming PRs will add the functionality you suggest.

**Pull request merged**